src/net/sourceforge/zmanim/ComplexZmanimCalendar.java

```diff
@@ -2706,4 +2736,278 @@ public Date getSolarMidnight() {
+	 */
+	public Date getSofZmanTfilaBaalHatanya() {
+		return getSofZmanTfila(getSunriseBaalHatanya(), getSunsetBaalHatanya());
+	}
+
+	/**
+	 * This method returns the latest time one is allowed eating chametz on Erev Pesach according to the opinion of the
+	 * <em>Baal Hatanya</em>. This time is identical to the {@link #getSofZmanTfilaBaalHatanya() Sof zman
+	 * tefilah Baal Hatanya}. This time is 4 hours into the day based on the opinion of the <em>Baal
+	 * Hatanya</em> that the day is calculated from sunrise to sunset. This returns the time 4 *
+	 * {@link #getShaahZmanisBaalHatanya()} after {@link #getSunriseBaalHatanya() <em>netz amiti</em> (sunrise)}.
+	 * 
+	 * @see #getShaahZmanisBaalHatanya()
+	 * @see #getSofZmanTfilaBaalHatanya()
+	 * @return the <code>Date</code> one is allowed eating chametz on Erev Pesach. If the calculation can't be computed
+	 *         such as in the Arctic Circle where there is at least one day a year where the sun does not rise, and one
+	 *         where it does not set, a null will be returned. See detailed explanation on top of the
+	 *         {@link AstronomicalCalendar} documentation.
+	 */
+	public Date getSofZmanAchilasChametzBaalHatanya() {
+		return getSofZmanTfilaBaalHatanya();
+	}
+
+	/**
+	 * This method returns the latest time for burning chametz on Erev Pesach according to the opinion of the
+	 * <em>Baal Hatanya</em>. This time is 5 hours into the day based on the opinion of the
+	 * <em>Baal Hatanya</em> that the day is calculated from sunrise to sunset. This returns the
+	 * time 5 * {@link #getShaahZmanisBaalHatanya()} after {@link #getSunriseBaalHatanya() <em>netz amiti</em> (sunrise)}.
+	 * 
+	 * @see #getShaahZmanisBaalHatanya()
+	 * @return the <code>Date</code> of the latest time for burning chametz on Erev Pesach. If the calculation can't be
+	 *         computed such as in the Arctic Circle where there is at least one day a year where the sun does not rise,
+	 *         and one where it does not set, a null will be returned. See detailed explanation on top of the
+	 *         {@link AstronomicalCalendar} documentation.
+	 */
+	public Date getSofZmanBiurChametzBaalHatanya() {
+		return getTimeOffset(getSunriseBaalHatanya(), getShaahZmanisBaalHatanya() * 5);
+	}
+
+	/**
+	 * This method returns <em>chatzos</em> (midday) following the opinion of the Baal Hatanya that the day for Jewish halachic
+	 * times start at {@link #getSunriseBaalHatanya() <em>netz amiti</em> (sunrise)} and ends at {@link #getSunsetBaalHatanya() 
+	 * <em>shkiah amiti</em> (sunset)}.
+	 * 
+	 * @see AstronomicalCalendar#getSunTransit()
+	 * @return the <code>Date</code> of chatzos. If the calculation can't be computed such as in the Arctic Circle where
+	 *         there is at least one day where the sun does not rise, and one where it does not set, a null will be
+	 *         returned. See detailed explanation on top of the {@link AstronomicalCalendar} documentation.
+	 */
+	public Date getChatzosBaalHatanya() {
```

Comment:
@yparitcher 
This method is not needed. Since the sunrise and sunset offset are identical, chatzos will be at the exact same time as the regular chatzos method.

Reply:
i originaly thought the same, however when i tested it, the chatzos times are between 200 ms and one second apart depending on location, date, and which calculator is used. therefore i added it for complete acuracy, although you are right that most users will never tell the difference.

Reply:
@yparitcher the minor differences for getChatzosBaalHatanya is due to rounding and are really irrelevant. They will only confuse. Please also update the minor conflict (due to a recent change that I introduced - sorry).
Thanks